### PR TITLE
feat: make `settings` optional for `metadata_properties`

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

This PR introduces the possibility to define `metadata_properties` without `settings` i.e. without a specific values or range of values to validate.

This PR affects both the API and the Python client.

**Type of change**

- [X] New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

- [X] Add unit tests for `Remote{Terms,Integer,Float}MetadataProperty` to ensure that `None` (optional) values are handled properly
- [X] Add unit tests for `{Terms,Integer,Float}MetadataProperty` to ensure that `None` (optional) values are handled properly
- [X] Add unit tests for the API to ensure that `None` (optional) values are handled properly

**Checklist**

- [ ] I added relevant documentation
- [X] I followed the style guidelines of this project
- [X] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [X] I have added relevant notes to the `CHANGELOG.md` file (See https://keepachangelog.com/)

### CODE REVIEW

File: docker/scripts/load_data.py [hunk 1]
```diff
@@ -195,7 +195,7 @@ def build_error_analysis_record(
         return rg.FeedbackRecord(fields=fields, metadata=metadata)
 
     @staticmethod
-    def load_error_analysis():
+    def load_error_analysis(with_settings_options: bool = True):
```
Comment (alvarobartt) [inline docker/scripts/load_data.py]:
```suggestion
    def load_error_analysis(with_metadata_property_options: bool = True):
```

File: docker/scripts/load_data.py [hunk 2]
```diff
@@ -217,19 +217,34 @@ def load_error_analysis():
...
-        metadata = [
-            rg.TermsMetadataProperty(name="correctness-langsmith", values=df.correctness_langsmith.unique().tolist()),
-            rg.TermsMetadataProperty(name="model-name", values=df.model_name.unique().tolist()),
-            rg.FloatMetadataProperty(name="temperature", min=df.temperature.min(), max=df.temperature.max()),
-            rg.IntegerMetadataProperty(name="max-tokens", min=df.max_tokens.min(), max=df.max_tokens.max()),
-            rg.FloatMetadataProperty(name="cpu-user", min=df.cpu_time_user.min(), max=df.cpu_time_user.max()),
-            rg.FloatMetadataProperty(name="cpu-system", min=df.cpu_time_system.min(), max=df.cpu_time_system.max()),
-            rg.TermsMetadataProperty(name="library-version", values=df.library_version.unique().tolist()),
-        ]
+        dataset_name = "error-analysis-with-feedback-alpha"
```
Comment (alvarobartt) [inline docker/scripts/load_data.py]: I think `-alpha` can now be safely removed

```suggestion
        dataset_name = "error-analysis-with-feedback"
```

File: src/argilla/client/feedback/schemas/metadata.py [hunk 3]
```diff
@@ -26,6 +26,7 @@
     root_validator,
     validator,
 )
+from pydantic.generics import GenericModel
```
Comment (alvarobartt) [inline src/argilla/client/feedback/schemas/metadata.py]: Unused import

```suggestion
```

File: src/argilla/client/feedback/schemas/metadata.py [hunk 4]
```diff
@@ -13,7 +13,7 @@
 #  limitations under the License.
 
 from abc import ABC, abstractmethod
-from typing import Any, Callable, Dict, List, Optional, Tuple, Union
+from typing import Any, Callable, Dict, Generic, List, Optional, Tuple, TypeVar, Union
```
Comment (alvarobartt) [inline src/argilla/client/feedback/schemas/metadata.py]: Unused imports

```suggestion
from typing import Any, Callable, Dict, List, Optional, Tuple, Union
```

**Review by alvarobartt — CHANGES_REQUESTED** [review]

File: src/argilla/client/feedback/schemas/metadata.py (hunk 3 above)
Comment (frascuchon) [reply src/argilla/client/feedback/schemas/metadata.py]: Mmmmm. The pre-commit step should remove them

Comment (gabrielmbmb) [reply src/argilla/client/feedback/schemas/metadata.py]: Right now `ruff` is not configured to remove unused imports, once we merge the metadata feature branch we could try to merge this one https://github.com/argilla-io/argilla/pull/3478 that adds some more linting rules, including the one of the unused imports

**Review by alvarobartt — APPROVED** [review]
